### PR TITLE
Change is_range_del_table_empty_ flag to atomic

### PR DESCRIPTION
**Pull request by abhimadan**
Summary: To avoid a race on the flag, make it an atomic_bool. This
doesn't seem to significantly affect benchmarks.

Test Plan:
- run readwhilewriting benchmark with db_bench on a DB with
range tombstones
- run crashtests with tsan enabled

### CODE REVIEW

File: db/memtable.h [hunk 1]
```diff
@@ -409,7 +409,7 @@ class MemTable {
   ConcurrentArena arena_;
   std::unique_ptr<MemTableRep> table_;
   std::unique_ptr<MemTableRep> range_del_table_;
-  bool is_range_del_table_empty_;
+  std::atomic_bool is_range_del_table_empty_;
```
Comment (miasantreble) [inline db/memtable.h]: looks like we use std::atomic<bool> much more often in the codebase, any idea if there is any difference?

Comment (abhimadan) [reply db/memtable.h]: It's typedef'd to `std::atomic<bool>` (see "Type Aliases" section on https://en.cppreference.com/w/cpp/atomic/atomic). I can change it for consistency if you want though.

**Comment by abhimadan** [comment]
Note: this fix is mainly to fix CI annoyances. In practice, this can only happen between a concurrent reader and writer, where the writer issues a DeleteRange that is about to complete as the read is processed. We know the write hasn't completed yet because `last_sequence_` hasn't been updated (which has acquire-release ordering that would prevent the flag from being incorrectly read), so we don't make any guarantees about what the reader should see.

**Comment by ajkr** [comment]
Thanks for writing down the details.

**Comment by abhimadan** [comment]
Before landing I'm going to change the memory ordering to `memory_order_relaxed`, since as mentioned, this is only to fix CI annoyances and not because the flag can be incorrectly read in practice.

**Pull request closed without merging**